homeassistant/components/matter/api.py

```diff
@@ -28,6 +39,40 @@ def async_register_api(hass: HomeAssistant) -> None:
     websocket_api.async_register_command(hass, websocket_commission_on_network)
     websocket_api.async_register_command(hass, websocket_set_thread_dataset)
     websocket_api.async_register_command(hass, websocket_set_wifi_credentials)
+    websocket_api.async_register_command(hass, websocket_node_diagnostics)
+    websocket_api.async_register_command(hass, websocket_ping_node)
+    websocket_api.async_register_command(hass, websocket_open_commissioning_window)
+    websocket_api.async_register_command(hass, websocket_remove_matter_fabric)
+    websocket_api.async_register_command(hass, websocket_interview_node)
+
+
+def async_get_node(
+    func: Callable[
+        [HomeAssistant, ActiveConnection, dict[str, Any], MatterAdapter, MatterNode],
+        Coroutine[Any, Any, None],
+    ],
+) -> Callable[
+    [HomeAssistant, ActiveConnection, dict[str, Any], MatterAdapter],
+    Coroutine[Any, Any, None],
+]:
+    """Decorate async function to get node."""
+
+    @wraps(func)
+    async def async_get_node_func(
+        hass: HomeAssistant,
+        connection: ActiveConnection,
+        msg: dict[str, Any],
+        matter: MatterAdapter,
+    ) -> None:
+        """Provide user specific data and store to function."""
+        node = node_from_ha_device_id(hass, msg[DEVICE_ID])
+        if not node:
```

Comment:
Side note:I think we also need to handle not loaded config entry. We don't need to fix that now though. It's an existing problem.

Here's what we do for Z-Wave:
https://github.com/home-assistant/core/blob/182d00be6638e76e49b3ec0d91a6507ce2f91fd7/homeassistant/components/zwave_js/api.py#L268-L286

Reply:
OK, added that as well but was not sure how to test it as I cant even create a fake device with an invalid config entry id

Reply:
sorry, only now see your comment that it didn't need to be fixed right away (its getting late haha) - well its now added, except for a specific test

Reply:
To test a missing config entry, we can create a device from another mock config entry and try to use that device as input.

To test a not loaded config entry, we can just unload the config entry and try the command again.

If it's late, I suggest we remove the config entry error handling for now and add that later with tests.